src/ray/pubsub/publisher.h

```diff
@@ -38,20 +38,42 @@ using SubscriberID = UniqueID;
 
 namespace pub_internal {
 
+class SubscriberState;
+
+/// State for an entity / topic in a pub/sub channel.
+struct EntityState {
```

Comment:
I can convert this struct to a class if anyone prefers.

Reply:
I'm ok with this.

Reply:
@mwtian really sorry for this, but I changed my mind after reviewing this PR again. :( :( 

I previously thought the member of this class will be used in other places.

It seems that pending_messages/message_sizes/total_size are actually only used within this structure. Why don't we just make it a class and make them private and move the `PublishToEntity` into this one?